src/rules/export.js

```diff
@@ -39,17 +38,22 @@ const tsTypePrefix = 'type:';
 function isTypescriptFunctionOverloads(nodes) {
   const nodesArr = Array.from(nodes);
 
-  const idents = flatMap(
-    nodesArr,
-    (node) => node.declaration && (
-      node.declaration.type === 'TSDeclareFunction' // eslint 6+
-      || node.declaration.type === 'TSEmptyBodyFunctionDeclaration' // eslint 4-5
-    )
-      ? node.declaration.id.name
-      : [],
-  );
-  if (new Set(idents).size !== idents.length) {
-    return true;
+  if (nodesArr[0].type === 'ExportDefaultDeclaration') {
```

Comment:
wait, why are we only checking the first item? what if it's a mix of an overload and something different?

Reply:
```
export { x as default };
export default function a() {}
```
Oh, I hadn't thought of this example before.
But TS will complain in this case, do we need to deal with it? If so I will update this PR.

Reply:
Yes, we still should deal with it.